### PR TITLE
Fix waveform overview lenght. 

### PR DESCRIPTION
**Pull request by daschuer**
This aims to fix https://github.com/mixxxdj/mixxx/issues/11344 where the waveform overview is rendered with the wrong length.
A possible regression from https://github.com/mixxxdj/mixxx/pull/11162 
[mixxx-2.3.3-145-gcfcd265afd-win64.msi](http://downloads.mixxx.org/snapshots/2.3/mixxx-2.3.3-145-gcfcd265afd-win64.msi)

### CODE REVIEW

**Comment by robbert-vdh** [comment]
I rebased this on top of the main branch and I can confirm that this also fixes the problem I posted about in https://github.com/mixxxdj/mixxx/issues/11280#issuecomment-1455146100.

File: src/widget/woverview.h
Comment (Swiftb0y) [inline src/widget/woverview.h:86]: IMO this only fixes the symptom of a more complex issue. If there is no track loaded, the CO should already be `0.0`...

Comment (daschuer) [reply src/widget/woverview.h:86]: Do you have a better solution? 
The complex issue is that the track change is propagated via the Qt signal queue while the track samples control is changed instantly as an atomic. 
m_trackLoaded bool ensures that the value is not used during the transition.

Comment (Swiftb0y) [reply src/widget/woverview.h:86]: I see, so its basically a data-race.
The problem I still have with this solution is that the waveform assumes that CO values `<= 0.0` mean that no track is loaded. Or rather, it relies on the fact that that is always the case (which it isn't due to the data-race). I think it would be better to move the `m_trackLoaded != nullptr` check to callers where they should actually check that.

Comment (daschuer) [reply src/widget/woverview.h:86]: The track length is set here to an invalid value from the worker thread: 
https://github.com/mixxxdj/mixxx/blob/ceaedb8443714459bc3544faab957d6903eba7b3/src/engine/enginebuffer.cpp#L526
in a responds from a new track loaded into the player. 

The Waveform overview visualizes the track loading process. It has already the new track before the engine had evicted the old one. In this situation is guarded by the m_trackLoaded boolean. 

I will add a comment.

Comment (daschuer) [reply src/widget/woverview.h:86]: I think a outdated waveform update signal cause a paint event with these inconstant data.

**Comment by daschuer** [comment]
@Swiftb0y done.

File: src/widget/woverview.h
Comment (Swiftb0y) [inline src/widget/woverview.h:85]:
```suggestion
            // Ignore the value, because the engine can still have the old track
            // during loading
```

**Review by Swiftb0y — COMMENTED** [review]
typo

File: src/widget/woverview.h
Comment (ronso0) [reply src/widget/woverview.h:85]: hmm, codespell didn't catch that

Comment (daschuer) [reply src/widget/woverview.h]: I have amended the last commit

**Comment by daschuer** [comment]
Done

**Comment by ronso0** [comment]
Two users confirmed this fixes #11344 so this is ready to go without a second review?

**Review by Swiftb0y — APPROVED** [review]
Sure

**Pull request merged**
